static/app/stories/view/storySearch.tsx

```diff
@@ -149,19 +149,25 @@ function filter(textValue: string, inputValue: string): boolean {
   return match.score > 0;
 }
 
-function SearchComboBox<T extends StoryTreeNode>(props: SearchComboBoxProps<T>) {
+function SearchComboBox(props: SearchComboBoxProps) {
   const [inputValue, setInputValue] = useState('');
   const {inputRef} = props;
   const listBoxRef = useRef<HTMLUListElement | null>(null);
   const popoverRef = useRef<HTMLDivElement | null>(null);
   const navigate = useNavigate();
   const handleSelectionChange = useCallback(
     (key: Key | null) => {
-      if (key) {
-        navigate(`/stories?name=${key}`, {replace: true});
+      if (!key) {
+        return;
       }
+      const node = getStoryTreeNodeFromKey(key, props);
+      if (!node) {
+        return;
+      }
+      const {state, ...to} = node.location;
+      navigate(to, {replace: true, state});
     },
-    [navigate]
+    [navigate, props]
```

Comment:
`props` will be a new object on every render, so adding it to `useCallback` nullifies the memoization. I think we should just remove `useCallback` here altogether, as we only pass it to `onSelectionChange` of `useComboBoxState` and that doesn’t require memoization.